stellargraph/layer/node2vec.py

```diff
@@ -0,0 +1,204 @@
+            )
+
+        self.multiplicity = generator.multiplicity
+        self.input_node_num = generator.graph.number_of_nodes()
+
+        if len(list(generator.graph.node_types)) > 1:
+            raise RuntimeError("Node2Vec called on graph with more than one node type.")
+
+    def _get_sizes_from_keywords(self, kwargs):
+        """
+        Sets node_num and multiplicity from the keywords.
+        Args:
+             kwargs: The additional keyword arguments.
+        """
+        try:
+            self.input_node_num = kwargs["node_num"]
+            self.multiplicity = kwargs["multiplicity"]
+
+        except KeyError:
+            raise KeyError(
+                "Generator not provided; node_num and multiplicity must be specified."
+            )
+
+    def __call__(self, xin, embedding):
+        """
+        Construct node representations from node ids through a look-up table.
+
+        Args:
+            xin (Keras Tensor): Batch input node ids.
+            embedding (str): "input" for input_embedding matrix, "output" for output_embedding
+
+        Returns:
+            Output tensor.
+        """
+
+        if embedding == "input":
+            h_layer = self.input_embedding(xin)
+        elif embedding == "output":
+            h_layer = self.output_embedding(xin)
+        else:
+            raise ValueError(
+                'wrong embedding argument is supplied: {}, should be "input" or "output"'.format(
+                    embedding
+                )
+            )
+
+        h_layer = Reshape((self.emb_size,))(h_layer)
+        # K.squeeze(h_layer, axis=0)
+
+        return h_layer
+
+    def node_model(self, embedding="input"):
+        """
+        Builds a Node2Vec model for node prediction.
+
+        Args:
+            embedding (str): "input" for input_embedding, "output" for output_embedding
+
+        Returns:
+            tuple: (x_inp, x_out) where ``x_inp`` is a Keras input tensor
+            for the Node2Vec model and ``x_out`` is the Keras tensor
+            for the Node2Vec model output.
+
+        """
+        # Create tensor inputs
+        x_inp = Input(shape=(1,))
+
+        # Output from Node2Vec model
+        x_out = self(x_inp, embedding)
+
+        return x_inp, x_out
+
+    def link_model(self):
+        """
+        Builds a Node2Vec model for link or node pair prediction.
+
+        Returns:
+            tuple: (x_inp, x_out) where ``x_inp`` is a list of Keras input tensors for (src, dst) nodes in the node pairs
+            and ``x_out`` is a list of output tensors for (src, dst) nodes in the node pairs.
+
+        """
+        # Expose input and output sockets of the model, for source node:
+        x_inp_src, x_out_src = self.node_model("input")
+        x_inp_dst, x_out_dst = self.node_model("output")
+
+        x_inp = [x_inp_src, x_inp_dst]
+        x_out = [x_out_src, x_out_dst]
+        return x_inp, x_out
+
+    def build(self):
```

Comment:
We ended up deprecating this method in https://github.com/stellargraph/stellargraph/pull/1140 - given that this is an entirely new model, maybe we don't need old methods with deprecation warnings either, such as the `default_model` method below? So in summary:

* `node_model` -> `_node_model` (i.e. should be private)
* `link_model` -> `_link_model` (i.e. should be private)
* `build` -> `in_out_tensors`
* remove `default_model`